lib/m2ts/ts-correction-stream.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * mux.js
+ *
+ * Copyright (c) 2016 Brightcove
+ * All rights reserved.
+ *
+ * Accepts program elementary stream (PES) data events and corrects
+ * decode and presentation time stamps to account for a rollover
+ * of the 33 bit value.
+ */
+ 
+'use strict';
+
+var Stream = require('../utils/stream');
+
+var MAX_TS = 8589934592;
+
+var RO_THRESH = 4294967296;
+
+var TSCorrectionStream = function() {
+
+  var
+    reference = {
+      video: {
+        lastDTS: undefined,
+        referenceDTS: undefined
+      },
+      audio: {
+        lastDTS: undefined,
+        referenceDTS: undefined
+      },
+      'timed-metadata': {
+        lastDTS: undefined,
+        referenceDTS: undefined
+      }
+    };
+
+  TSCorrectionStream.prototype.init.call(this);
+
+  var handleRollover = function(value, reference) {
+    var direction = 1;
+
+    if (value > reference) {
```

Comment:
I think this might need a comment explaining the idea behind this logic and why `value > reference` means `direction = -1`. It took me a few moments to get it and I more or less knew what to expect!